ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriter.scala

```diff
@@ -112,19 +132,30 @@ object SqlLedgerReaderWriter {
           updateOrRetrieveLedgerId(ledgerId, database).removeExecutionContext
         )
         dispatcher <- new DispatcherOwner(database).acquire()
-        validator = SubmissionValidator.createForTimeMode(
-          ledgerStateAccess = new SqlLedgerStateAccess(database, metrics),
-          logEntryIdAllocator = logEntryIdAllocator,
-          checkForMissingInputs = false,
-          stateValueCache = stateValueCache,
-          engine = engine,
-          metrics = metrics,
-          inStaticTimeMode = timeProvider != TimeProvider.UTC,
+        sqlLedgerStateAccess = new SqlLedgerStateAccess(database, metrics)
+        keyValueCommitting = new KeyValueCommitting(
+          engine,
+          metrics,
         )
-        committer = new ValidatingCommitter[Index](
-          () => timeProvider.getCurrentTime,
-          validator = validator,
-          postCommit = dispatcher.signalNewHead,
```

Comment:
Looks like we will never call `dispatcher.signalNewHead` any more. In the in-memory ledger this call is done after `commit`, but I'm not sure that's possible here.

You may need to adjust `PreExecutingValidatingCommitter` and `RawPostExecutionWriter` to return a tuple of `(Index, SubmissionResult)`. (The index can be retrieved from `LedgerStateOperations#appendToLog`.)

This should also simplify the in-memory code; you'll be able to delete the `InMemoryState#newHeadSinceLastWrite` method.

Reply:
Got it, thanks for the explanation.